crates/wasmtime/src/config.rs

```diff
@@ -1269,6 +1269,15 @@ impl Config {
         self
     }
 
+    /// Configures whether extra debug checks are inserted into
+    /// Wasmtime-generated code by Cranelift.
+    ///
+    /// The default value for this is `false`
+    #[cfg(any(feature = "cranelift", feature = "winch"))]
```

Comment:
Should this be enabled only for `feature = cranelift` and not `feature = winch` also?

If we do want this flag for winch as well, perhaps we should rename the method to not have `cranelift` in its name? Maybe `compiled_code_debug_checks`?

Reply:
Sort of, it's kind of complicated. The genesis of this particular `#[cfg]` is a copy/paste from the above function which suffers the same problem you're describing. However it's also correct in that it's only affecting Cranelift-generated code, not Winch-generated code. But it's also true that when you use Winch you always use Cranelift as well (due to trampoline generation being entirely in Cranelift).

So in a sense this is actually accurate, for either compiler configuration this only affects Cranelift. It's all a bit confusing though and my general hope is that this is just some esoteric thing that most won't run into.

I'll spin this out into https://github.com/bytecodealliance/wasmtime/issues/11610 though